spec/01-unit/03-conf_loader_spec.lua

```diff
@@ -824,6 +824,36 @@ describe("Configuration loader", function()
           assert.is_nil(err)
           assert.same(nil, conf.ssl_ciphers)
         end)
+        it("defines ssl_dhparam with default cipher suite", function()
+          local conf, err = conf_loader()
+          assert.is_nil(err)
+          assert.equal("ffdhe2048", conf.nginx_http_ssl_dhparam)
+          assert.equal("ffdhe2048", conf.nginx_stream_ssl_dhparam)
+        end)
+        it("defines ssl_dhparam with intermediate cipher suite", function()
+          local conf, err = conf_loader(nil, {
+            ssl_cipher_suite = "intermediate",
+          })
+          assert.is_nil(err)
+          assert.equal("ffdhe2048", conf.nginx_http_ssl_dhparam)
+          assert.equal("ffdhe2048", conf.nginx_stream_ssl_dhparam)
+        end)
+        it("doesn't define ssl_dhparam with modern cipher suite", function()
+          local conf, err = conf_loader(nil, {
+            ssl_cipher_suite = "modern",
+          })
+          assert.is_nil(err)
+          assert.equal(nil, conf.nginx_http_ssl_dhparam)
+          assert.equal(nil, conf.nginx_stream_ssl_dhparam)
+        end)
+        it("doesn't define ssl_dhparam with old cipher suite (#todo)", function()
+          local conf, err = conf_loader(nil, {
+            ssl_cipher_suite = "old",
+          })
+          assert.is_nil(err)
+          assert.equal(nil, conf.nginx_http_ssl_dhparam)
+          assert.equal(nil, conf.nginx_stream_ssl_dhparam)
+        end)
```

Comment:
We can also add a check for file's existence and the `abspath` transformation like https://github.com/Kong/kong/pull/6470/files#diff-ecda8b9e752563281567542698215736bfc851121ddd9cca53621f1ee8cd755bR882.

The file does not need to point to real dhparams, we only care about the config string being correctly expanded.